src/c++/perf_analyzer/client_backend/mock_client_backend.h

```diff
@@ -35,6 +35,38 @@
 
 namespace triton { namespace perfanalyzer { namespace clientbackend {
 
+/// Mock class of an InferInput
+///
+class MockInferInput : public InferInput {
+ public:
+  MockInferInput(
+      const BackendKind kind, const std::string& name,
+      const std::vector<int64_t>& dims, const std::string& datatype)
+      : InferInput(kind, name, datatype), dims_(dims)
+  {
+  }
+
+  const std::vector<int64_t>& Shape() const override { return dims_; }
+
+  Error Reset() override { return cb::Error::Success; }
+
+  Error AppendRaw(const uint8_t* input, size_t input_byte_size) override
+  {
+    recorded_inputs_.push_back(input);
```

Comment:
This would push back the pointer, right?  Seems like you need to get the real data out.

Reply:
It's a pointer to the real input data. It remains valid after the scope of this function, so this should be fine.

Reply:
You lose the size information though? I don't expect we are going to be testing with a bunch of different sizes, but it seems like this would be the safest point to convert pointers to actual data.

Reply:
I don't think I can convert to actual data here without it getting complex. I can record the sizes though.

Reply:
How is it any different than the casting you are doing in the CHECK?

Reply:
Added the sizes and verified those.